Video-Transformers/Video-Transformers/ViewController.swift

```diff
@@ -0,0 +1,266 @@
+     * to the OpenTok session.
+     */
+    fileprivate func doPublish() {
+        var error: OTError?
+        defer {
+            processError(error)
+        }
+        
+        session.publish(publisher, error: &error)
+        
+        if let pubView = publisher.view {
+            pubView.frame = CGRect(x: 0, y: 0, width: kWidgetWidth, height: kWidgetHeight)
+            view.addSubview(pubView)
+        }
+        
+        // Configure toogle button
+        buttonVideoTransformerToggle = UIButton(type: .custom)
+        buttonVideoTransformerToggle.frame = CGRect(x: kWidgetWidth - 65, y: 50, width: 50, height: 25)
+        buttonVideoTransformerToggle.layer.cornerRadius = 5.0
+        self.view.addSubview(buttonVideoTransformerToggle)
+        self.view.bringSubviewToFront(buttonVideoTransformerToggle)
+        buttonVideoTransformerToggle.setTitle("set", for: .normal)
+        buttonVideoTransformerToggle.titleLabel?.font = UIFont.systemFont(ofSize: 12)
+        buttonVideoTransformerToggle.setTitleColor(.gray, for: .normal)
+        buttonVideoTransformerToggle.backgroundColor = .white
+        buttonVideoTransformerToggle.layer.borderWidth = 1.0
+        buttonVideoTransformerToggle.layer.borderColor = UIColor.gray.cgColor
+        buttonVideoTransformerToggle.addTarget(self, action: #selector(buttonTapped(_:)), for: .touchUpInside)
+
+    }
+    
+    /**
+     * Instantiates a subscriber for the given stream and asynchronously begins the
+     * process to begin receiving A/V content for this stream. Unlike doPublish,
+     * this method does not add the subscriber to the view hierarchy. Instead, we
+     * add the subscriber only after it has connected and begins receiving data.
+     */
+    fileprivate func doSubscribe(_ stream: OTStream) {
+        var error: OTError?
+        defer {
+            processError(error)
+        }
+        subscriber = OTSubscriber(stream: stream, delegate: self)
+        
+        session.subscribe(subscriber!, error: &error)
+    }
+    
+    fileprivate func cleanupSubscriber() {
+        subscriber?.view?.removeFromSuperview()
+        subscriber = nil
+    }
+    
+    fileprivate func cleanupPublisher() {
+        publisher.view?.removeFromSuperview()
+    }
+    
+    fileprivate func processError(_ error: OTError?) {
+        if let err = error {
+            DispatchQueue.main.async {
+                let controller = UIAlertController(title: "Error", message: err.localizedDescription, preferredStyle: .alert)
+                controller.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
+                self.present(controller, animated: true, completion: nil)
+            }
+        }
+    }
+    
+    var logoTransformer: CustomTransformer = CustomTransformer() // Create an instance of CustomTransformer
+    
+    var isSet = false
+
+    @objc func buttonTapped(_ sender: UIButton) {
+        if !isSet {
```

Comment:
`isSet` sounds too generic and unnecessary, maybe you can check `publisher.videoTransformers.isEmpty`.

Reply:
You are right. Changing it